src/waveform/renderers/allshader/waveformrendererendoftrack.h

```diff
@@ -34,11 +36,20 @@ class allshader::WaveformRendererEndOfTrack final
     // Virtual for rendergraph::Node
     void preprocess() override;
 
+  public slots:
+    void setColor(const QColor& color) {
+        m_color = color;
+    }
+    void setEndOfTrackWarningTime(int endOfTrackWarningTime) {
+        m_remainingTimeTriggerSeconds = endOfTrackWarningTime;
+    }
+
   private:
     std::unique_ptr<ControlProxy> m_pEndOfTrackControl;
     std::unique_ptr<ControlProxy> m_pTimeRemainingControl;
 
     QColor m_color;
+    int m_remainingTimeTriggerSeconds;
```

Comment:
I don't see this getting initialized anywhere. Am I missing something?

Reply:
Nope, it's missing indeed